### PR TITLE
[Tommy Collison] Sprint Challenge: Node Express

### PR DESCRIPTION
**Pull request by tommycollison**
🥇 🥈@John-Spraul

### CODE REVIEW

**Comment by tommycollison** [comment]
Pretty sure all the questions are correct minus the last one.

File: Review.md [hunk 1]
```diff
@@ -2,16 +2,32 @@
...
 
+These are functions that get the request and response objects and can operate on them before passing them on to the next part of the process. Useful for logging, or authentication.
+
 ## What is a Resource?
 
+Basically everything in Node is a resource. Anything piece of data that's accessible through a unique URL.
+
 ## What can the API return to help clients know if a request was successful?
 
+An error message, or the data they requested.
```
Comment (John-Spraul) [inline Review.md]: Could include the HTTP status codes here as well

File: Review.md [hunk 2]
```diff
@@ -2,16 +2,32 @@
...
+
 ## How can we partition our application into sub-applications?
 
+Routers!
+
 ## What is express.json() and why do we need it?
+
+These are what allow us to create middleware.
```
Comment (John-Spraul) [inline Review.md]: express.json() is middleware that lets us use JS objects in our requests and responses

File: server.js [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+const server = express(); // creates the server
+
+const router = express.Router();
+const projectsRouter = require('./projects');
+const actionsRouter = require('./actions')
+const actionz = require('./data/helpers/actionModel')
+const projectz = require('./data/helpers/projectModel')
+
+
+server.use(express.json());
```
Comment (John-Spraul) [inline server.js]: Remember that the order is important ⚠️

File: server.js [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        res.json(`Huh, can't find those projects.`)
+    })
+})
+
+// get a specific action 
+
+server.get('/api/actions/:id', (req, res) => {
+    const {id} = req.params;
+    actionz.get(id)
+    .then(user => {
```
Comment (John-Spraul) [inline server.js]: user is a poor choice for a placeholder here

File: server.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        .json('Error: failed to add project')
+    })
+})
+
+
+// delete project
+
+server.delete('/api/projects/:id', (req, res) => {
+    const {id} = req.params;
+    const project = req.body;
```
Comment (John-Spraul) [inline server.js]: A delete endpoint generally doesn't make use of the req.body

**Pull request closed without merging**